infrastructure/src/main/java/org/corfudb/infrastructure/log/SegmentManager.java

```diff
@@ -0,0 +1,56 @@
+package org.corfudb.infrastructure.log;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * A segment manger which manages stream and garbage segment mappings.
+ *
+ * Created by WenbinZhu on 5/28/19.
+ */
+@RequiredArgsConstructor
+public class SegmentManager {
+
+    private final StreamLogParams logParams;
+
+    @Getter
+    private final Map<Long, StreamLogSegment> streamLogSegments = new ConcurrentHashMap<>();
```

Comment:
More detailed explanation on the two variables, including what the key means.

Reply:
Yes, will be added once it's finalized.